Lib9c/Action/ArenaBattle.cs

```diff
@@ -0,0 +1,163 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Bencodex.Types;
+using Libplanet;
+using Libplanet.Action;
+using Nekoyume.Battle;
+using Nekoyume.Extensions;
+using Nekoyume.Helper;
+using Nekoyume.Model.Item;
+using Nekoyume.Model.State;
+using Nekoyume.TableData;
+using static Lib9c.SerializeKeys;
+
+namespace Nekoyume.Action
+{
+    [Serializable]
+    [ActionType("arena_battle")]
+    public class ArenaBattle : GameAction
+    {
+        public Address avatarAddress;
+        public Address enemyAddress;
+        public List<Guid> costumeIds;
+        public List<Guid> equipmentIds;
+        public int ticket;
+        public int arenaId;
+
+        protected override IImmutableDictionary<string, IValue> PlainValueInternal =>
+            new Dictionary<string, IValue>()
+            {
+                ["avatarAddress"] = avatarAddress.Serialize(),
+                ["enemyAddress"] = enemyAddress.Serialize(),
+                ["costume_ids"] = new List(costumeIds
+                    .OrderBy(element => element).Select(e => e.Serialize())),
+                ["equipment_ids"] = new List(equipmentIds
+                    .OrderBy(element => element).Select(e => e.Serialize())),
+            }.ToImmutableDictionary();
+
+        protected override void LoadPlainValueInternal(
+            IImmutableDictionary<string, IValue> plainValue)
+        {
+            avatarAddress = plainValue["avatarAddress"].ToAddress();
+            enemyAddress = plainValue["enemyAddress"].ToAddress();
+            costumeIds = ((List)plainValue["costume_ids"]).Select(e => e.ToGuid()).ToList();
+            equipmentIds = ((List)plainValue["equipment_ids"]).Select(e => e.ToGuid()).ToList();
+        }
+
+        public override IAccountStateDelta Execute(IActionContext context)
+        {
+            var states = context.PreviousStates;
+            var blockIndex = context.BlockIndex;
+            var arenaAvatarStateAddress = ArenaAvatarState.DeriveAddress(avatarAddress);
+            var arenaStateAddress = ArenaState.DeriveAddress(arenaId);
+            var inventoryAddress = avatarAddress.Derive(LegacyInventoryKey);
+            var questListAddress = avatarAddress.Derive(LegacyQuestListKey);
+
+            if (context.Rehearsal)
+            {
+                return states
+                    .SetState(arenaStateAddress, MarkChanged)
+                    .SetState(arenaAvatarStateAddress, MarkChanged)
+                    .SetState(avatarAddress, MarkChanged)
+                    .SetState(inventoryAddress, MarkChanged)
+                    .SetState(questListAddress, MarkChanged);
```

Comment:
We do not need to mark some addresses now. Please return the states as it is when rehearsal.

```suggestion
                return states;
```